pkg/operator/reports_test.go

```diff
@@ -92,3 +101,217 @@ func TestGetNextReportPeriod(t *testing.T) {
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.InvalidReportReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is true and reason is InvalidReport return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionTrue, util.InvalidReportReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is false and reason is RunImmediately return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.RunImmediatelyReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is true and reason is RunImmediately return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionTrue, util.RunImmediatelyReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+	}
+
+	for _, testCase := range testTable {
+		var mockLogger = logrus.New()
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			runningCond := isReportFinished(mockLogger, testCase.report)
+			assert.Equalf(t, runningCond, testCase.expectFinished, "expected the report '%s' would return '%t', but got '%t'", testCase.name, testCase.expectFinished, runningCond)
+		})
+	}
+}
+
+func TestValidateReport(t *testing.T) {
+	const (
+		testNamespace           = "default"
+		testReportName          = "test-report"
+		testQueryName           = "test-query"
+		testGenerationQueryName = "test-GQN"
+	)
+
+	ds1 := testhelpers.NewReportDataSource("datasource1", testNamespace)
+
+	testQuery := &metering.ReportGenerationQuery{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-query",
+			Namespace: testNamespace,
+		},
+		Spec: metering.ReportGenerationQuerySpec{
+			DataSources: []string{
+				"datasource1",
+			},
+		},
+	}
+
+	dataSourceGetter := testhelpers.NewReportDataSourceStore([]*metering.ReportDataSource{ds1})
+	queryGetter := testhelpers.NewReportGenerationQueryStore([]*metering.ReportGenerationQuery{testQuery})
+	reportGetter := testhelpers.NewReportStore(nil)
+
+	reportStart := &time.Time{}
+	reportEndTmp := reportStart.AddDate(0, 1, 0)
+	reportEnd := &reportEndTmp
+
+	testTable := []struct {
+		name         string
+		report       *metering.Report
+		queryName    string
```

Comment:
Based on my other review comments, this field should be removed. Instead, set the correct query name when you call `testhelper.NewReport`

Reply:
Oh yikes, I misinterpreted the testQueryName field in `NewReport`